### ISSUE TITLE
黄埔学院 x2paddle

### ISSUE THREAD

**Issue by yajiezhu** [PaddlePaddle/X2Paddle#382]
textCNN模型 tf2pd 转换的pd模型应该有两个文件夹 但是 我只有1个model_with_code 没有inference_model 还报错了x2paddle: error: unrecognized arguments: --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model

**Comment by jiangjiajun** [comment]
> textCNN模型 tf2pd 转换的pd模型应该有两个文件夹 但是 我只有1个model_with_code 没有inference_model 还报错了x2paddle: error: unrecognized arguments: --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model

这个命令行报错，是说明x2paddle没有安装成功的原因。

TextCNN模型是可以支持转换的，下载最新X2Paddle代码安装即可
```
pip uninstall x2paddle
git clone https://github.com/PaddlePaddle/X2Paddle.git
cd X2Paddle
python setup.py install
```

**Comment by yajiezhu** [comment]
我按照你说的 做了一遍 发现 效果一样 只有1个model_with_code 没有inference_model 还报错了x2paddle: error: unrecognized arguments: --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model

**Comment by jiangjiajun** [comment]
> 我按照你说的 做了一遍 发现 效果一样 只有1个model_with_code 没有inference_model 还报错了x2paddle: error: unrecognized arguments: --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model

麻烦再检查查您的环境哈，如果会报命令行出错 error: unrecognized arguments: --framework=tensorflow ，为什么还会有model_with_code这个目录生成呢

**Comment by yajiezhu** [comment]
卸载x2paddle的确成功了 安装也成功了

**Comment by jiangjiajun** [comment]
> 卸载x2paddle的确成功了 安装也成功了

删掉之前转换生成的文件，重新转换试下。另外你是Windows还是Linux呢

**Comment by yajiezhu** [comment]
之前转换生成的文件删掉了 重新转换了 linux
(base) zyj@ubuntu18:~/pythonProgram/receiveVisitorRobot$ x2paddle --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model
paddle.__version__ = 1.8.2
/home/zyj/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Now translating model from tensorflow to paddle.
Unknown shape for input tensor[tensor name: "dropout_keep_prob"]
Use your keyboard type the shape of input tensor below :)
Shape of Input(e.g. None,224,224,3): None,384
Total nodes: 52
Converting node 52 ...     
Done!
usage: x2paddle [-h] [--train_data TRAIN_DATA] [--train_file TRAIN_FILE]
                [--test_data TEST_DATA] [--test_file TEST_FILE]
                [--batch_size BATCH_SIZE] [--epoch EPOCH]
                [--hidden_dim HIDDEN_DIM] [--optimizer OPTIMIZER] [--CRF CRF]
                [--lr LR] [--clip CLIP] [--dropout DROPOUT]
                [--update_embedding UPDATE_EMBEDDING]
                [--pretrain_embedding PRETRAIN_EMBEDDING]
                [--embedding_dim EMBEDDING_DIM] [--shuffle SHUFFLE]
                [--mode MODE] [--demo_model DEMO_MODEL]
x2paddle: error: unrecognized arguments: --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model
(base) zyj@ubuntu18:~/pythonProgram/receiveVisitorRobot$

**Comment by jiangjiajun** [comment]
这里看转换时，输入的shape应该不是None,384, 而应该是1   因为这里要输入的是tensor: dropout_keep_prob的shape

**Comment by yajiezhu** [comment]
还是报一样的错误
(base) zyj@ubuntu18:~/pythonProgram/receiveVisitorRobot$ x2paddle --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model
paddle.__version__ = 1.8.2
/home/zyj/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Now translating model from tensorflow to paddle.
Unknown shape for input tensor[tensor name: "dropout_keep_prob"]
Use your keyboard type the shape of input tensor below :)
Shape of Input(e.g. None,224,224,3): 1
Total nodes: 52
Converting node 52 ...     
Done!
usage: x2paddle [-h] [--train_data TRAIN_DATA] [--train_file TRAIN_FILE]
                [--test_data TEST_DATA] [--test_file TEST_FILE]
                [--batch_size BATCH_SIZE] [--epoch EPOCH]
                [--hidden_dim HIDDEN_DIM] [--optimizer OPTIMIZER] [--CRF CRF]
                [--lr LR] [--clip CLIP] [--dropout DROPOUT]
                [--update_embedding UPDATE_EMBEDDING]
                [--pretrain_embedding PRETRAIN_EMBEDDING]
                [--embedding_dim EMBEDDING_DIM] [--shuffle SHUFFLE]
                [--mode MODE] [--demo_model DEMO_MODEL]
x2paddle: error: unrecognized arguments: --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model

**Comment by jiangjiajun** [comment]
> 还是报一样的错误
> (base) zyj@ubuntu18:~/pythonProgram/receiveVisitorRobot$ x2paddle --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model
> paddle.**version** = 1.8.2
> /home/zyj/anaconda3/lib/python3.6/site-packages/h5py/**init**.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
> from ._conv import register_converters as _register_converters
> Now translating model from tensorflow to paddle.
> Unknown shape for input tensor[tensor name: "dropout_keep_prob"]
> Use your keyboard type the shape of input tensor below :)
> Shape of Input(e.g. None,224,224,3): 1
> Total nodes: 52
> Converting node 52 ...
> Done!
> usage: x2paddle [-h] [--train_data TRAIN_DATA] [--train_file TRAIN_FILE]
> [--test_data TEST_DATA] [--test_file TEST_FILE]
> [--batch_size BATCH_SIZE] [--epoch EPOCH]
> [--hidden_dim HIDDEN_DIM] [--optimizer OPTIMIZER] [--CRF CRF]
> [--lr LR] [--clip CLIP] [--dropout DROPOUT]
> [--update_embedding UPDATE_EMBEDDING]
> [--pretrain_embedding PRETRAIN_EMBEDDING]
> [--embedding_dim EMBEDDING_DIM] [--shuffle SHUFFLE]
> [--mode MODE] [--demo_model DEMO_MODEL]
> x2paddle: error: unrecognized arguments: --framework=tensorflow --model=./resource/model.pb --save_dir=./paddle_model

usage: x2paddle [-h] [--train_data TRAIN_DATA] [--train_file TRAIN_FILE]
[--test_data TEST_DATA] [--test_file TEST_FILE]
[--batch_size BATCH_SIZE] [--epoch EPOCH]
[--hidden_dim HIDDEN_DIM] [--optimizer OPTIMIZER] [--CRF CRF]
[--lr LR] [--clip CLIP] [--dropout DROPOUT]
[--update_embedding UPDATE_EMBEDDING]
[--pretrain_embedding PRETRAIN_EMBEDDING]
[--embedding_dim EMBEDDING_DIM] [--shuffle SHUFFLE]
[--mode MODE] [--demo_model DEMO_MODEL]

这里这一段信息提示，在X2Paddle里面其实是没有这些东西相关的，怀疑你的环境有问题，建议重新创建Python环境安装X2Paddle

**Comment by yajiezhu** [comment]
python环境里有很多安装包 都要重新安装  有点搞大了 x2paddle 卸载再安装 也没有搞定 没有别的办法了吗

**Comment by yajiezhu** [comment]
我换了python环境 也不行 一样的错误

**Comment by jiangjiajun** [comment]
> 我换了python环境 也不行 一样的错误

由于我不了解您的环境具体情况，但当前出现的问题肯定是由于您的Python环境问题导致，在上面的提示中，出现的unrec等提示均与X2Paddle是无关的。我这边已经试过您之前发给我们的模型，参考如下截图，是可以正常转换完成的，建议检查自己的Python环境 ：）

![image](https://user-images.githubusercontent.com/19339784/90849131-26e4f380-e3a1-11ea-874c-32220bb5ea15.png)

**Comment by yajiezhu** [comment]
python环境 我的确换了 一样的报错 你的paddle版本是1.8.4 我的是1.8.2呢  这个有影响吗

**Comment by jiangjiajun** [comment]
跟paddle版本没关系，主要是由于你的出错提示出现了与X2Paddle无关的日志，所以环境上应该有问题。

你可以通过Anaconda创建一个新的Python环境安装paddlepaddle, tensorflow 和 x2paddle，然后再进行转换

**Comment by yajiezhu** [comment]
不是环境的问题 我已经换环境了 报同样的错误 现在搞好了 可以了 不过 我发给你的textCNN模型可以tf2pd 但是其他的textCNN模型是转换不行的 会报我现在遇到的问题

**Comment by jiangjiajun** [comment]
> 不是环境的问题 我已经换环境了 报同样的错误 现在搞好了 可以了 不过 我发给你的textCNN模型可以tf2pd 但是其他的textCNN模型是转换不行的 会报我现在遇到的问题

如果方便的话，你可以发一个你那边转换出错的模型给我，我来试下

**Comment by yajiezhu** [comment]
转换错的textCNN模型在附件里 请查收~




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/X2Paddle"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年8月21日(星期五) 中午1:54
收件人:&nbsp;"PaddlePaddle/X2Paddle"<X2Paddle@noreply.github.com&gt;;
抄送:&nbsp;"朱亚杰"<769877925@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [PaddlePaddle/X2Paddle] 黄埔学院 x2paddle (#382)





  
不是环境的问题 我已经换环境了 报同样的错误 现在搞好了 可以了 不过 我发给你的textCNN模型可以tf2pd 但是其他的textCNN模型是转换不行的 会报我现在遇到的问题
  
如果方便的话，你可以发一个你那边转换出错的模型给我，我来试下

**Comment by yajiezhu** [comment]
textCNN模型 邮件发给你了 你试一下

**Comment by jiangjiajun** [comment]
> textCNN模型 邮件发给你了 你试一下

麻烦直接发送至jiangjiajun@baidu.com

**Comment by yajiezhu** [comment]
textCNN模型 tf2pd 后 我想pd2lite  结果也报错了 你试一下
paddle_lite_opt --model_dir=./inference_model --valid_targets=arm --optimize_out_type=naive_buffer --optimize_out=./textCNN_model

**Comment by jiangjiajun** [comment]
> textCNN模型 tf2pd 后 我想pd2lite 结果也报错了 你试一下
> paddle_lite_opt --model_dir=./inference_model --valid_targets=arm --optimize_out_type=naive_buffer --optimize_out=./textCNN_model

paddleLite相关问题可在PaddleLite的github repo提issue哈

**Comment by yajiezhu** [comment]
ok 我转错的textCNN模型 你收到了吗

**Comment by jiangjiajun** [comment]
我这边仍然可以正常转换，还是建议你重新搭建一个python环境，X2Paddle不会有你先前粘贴的那些日志提示

![image](https://user-images.githubusercontent.com/19339784/90859792-6fa9a600-e3bb-11ea-80e6-62997652c3c5.png)

**Comment by yajiezhu** [comment]
我把转错的textCNN模型 放在了另外一个项目里 重新转换了一下 发现能转换成功 python环境并没有变
